test/widgets/action_sheet_test.dart

```diff
@@ -861,6 +862,51 @@ void main() {
           ..bodyFields['flag'].equals('read');
       });
     });
+
+    group('CopyTopicLinkButton', () {
+      setUp(() async {
+        TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+          SystemChannels.platform,
+          MockClipboard().handleMethodCall,
+        );
+      });
+
+      Future<void> tapCopyTopicLinkButton(WidgetTester tester) async {
+        await tester.ensureVisible(find.byIcon(ZulipIcons.link, skipOffstage: false));
+        await tester.tap(find.byIcon(ZulipIcons.link));
+        await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
+      }
+
+      group('copies topic link to clipboard', () {
+        testWidgets('FL >= 271 -> link contains "with" operator', (tester) async {
```

Comment:
nit: When there's a pair of tests for "legacy" vs. "current" behavior, I like to consider the "current" behavior as boring/default/unmarked, so for example naming the test like

```dart
testWidgets('copies message link to clipboard', (tester) async {
```

, and consider the "legacy" one as interesting/unusual/marked:

```dart
testWidgets('FL < 271 -> link doesn\'t contain "with" operator', (tester) async {
```


In the future, when we stop supporting servers with the legacy behavior,  "message link" will always include "with" in its meaning, and when we clear out the backwards-compatibility code and its corresponding test, the remaining test won't need any changes to make it look natural.